### ISSUE TITLE
Discussion for dealing with spam registrations

### ISSUE THREAD

**Issue by aseckin** [Metaculus/metaculus#1386]
A significant portion of new registrations share a pattern:
- They sign up, confirm email, skip Onboarding
- Go to profile, fill their bio with ad content that link to their sites
- Never come back to the site

From the session details I can tell that these are mostly real people and not bots. They don't really affect the site from a usage standpoint, but they skew product analytics and the db gets bloated. 

(Afaik, we disavow backlinks from profiles, so the spammers don't really get the benefit they're looking for - although they don't know that)

Potential next steps:
- Go through new registrations every day (about 100) and manually remove spam accounts
	- Maybe utilize an LLM to do the heavy-lifting?
- Disable markdown in user profiles in an attempt to disincentivize
- Do nothing

**Comment by SylvainChevalier** [comment]
Other ideas:

- Upon registration, allow forecasting and commenting, but you can only edit your bio after a week.
- Feed every bio to Claude, if it looks like an add send email to user that they have 24 hours to edit it or will be deactivated. Also give them email to appeal.
- Feed first N comments of every user to Claude. If it looks like an add: make it private, send email to user telling them how to appeal.

**Comment by aseckin** [comment]
Thanks for the clarification @draaglom I [created an issue](https://github.com/Metaculus/metaculus/issues/1421) for adding the ugc attribute to all links in comments and user profiles.

**Comment by SylvainChevalier** [comment]
Summary of chat we just had with @CodexVeritas:

- Focus on Bios.
- Everyday, run all bios that are less than a week old through LLM.
- Ones that are marked spam get an email warning them. Also includes way to appeal.
- If a bio is marked spam two days in a row: deactivate account, send email (with instructions to appeal again).

**Comment by SylvainChevalier** [comment]
Agreed about comments, but let's focus on bios as an mvp, since comments should be caught by users reporting them.

**Comment by CodexVeritas** [comment]
Also as a heads up, I think I'm going to approach this in a slightly different way. Rather than checking bios daily for a week, Spam is rejected at bio edit. This way people can't just wait a week and then give a spam bio. This also reduces the number of checks we have to do (since most bios edits will happen once from non-spammers, and the early message will probably stop human spammers). If multiple spam edits are detected, then the account is deactivated.

One concern though would be whether this would give a faster feedback loop to overcoming the spam filter (since there will likely be edge cases they could exploit).

Let me know if there are any thoughts to this approach.

**Comment by lsabor** [comment]
> I'm putting the true false-positive rate around 2-7%

I think this is too high. I'd rather let some spam through (it doesn't have much overall effect) than convict even one innocent. I think this should be improved and instead sacrifice false negatives.

>Spam detection is activated every profile update, and immediately deactivated if detected.

I'd prefer it to only activate on first profile update unless we later have good evidence to suggest it would be necessary. I can help implement that check since it might be subtle.

>integration tests

Linking your branch for my own reference: https://github.com/Metaculus/metaculus/tree/feat/spam-detection
Nice work! But yeah we'll need to look at that. I'm not sure why
Would you turn your branch into a PR and we can have a technical discussion there with code-comments etc?

>In the future, if I understand right, I normally shouldn't ask for approval at this step, but would just make a pull request to main (merge & squash style), and then mark this ticket as QA and @ someone for review?

So we've just shifted gears officially, and anything that isn't urgent needs a reviewer's approval now. I'd also suggest this being your first PR at Metaculus that it gets a good lookover.

> I'm going to test that the email is sent after deactivation once the merge is in production, since I couldn't do that locally.

I'd have to check, but I think either dev or play actually can send emails, but they automatically get captured and go to "engineering@metaculus.com" or something.

**Comment by CodexVeritas** [comment]
@lsabor Awesome! Let's move the discussion to the Pull Request. I'll start with a pull request for my second ticket.

**Comment by CodexVeritas** [comment]
@aseckin Check out the flow description in the pull request (https://github.com/Metaculus/metaculus/pull/1484)

**Comment by CodexVeritas** [comment]
@SylvainChevalier Spam detection was just merged. You should be able to make a new account, test a valid bio, have it update (as per normal), test an invalid bio, have your account deactivated, get a message, get logged out, and get an email.

**Comment by SylvainChevalier** [comment]
This has already reduced spam by ~4×. I was hoping for more, but this is already pretty good.

Some follow ups for improving record-keeping:

- Let's save the spam bios. Will help with understanding what is and is not detected.
- Let's record the spam detections. This can be a simple `spam_autodetected` field on the user.
